chain/events/events_called.go

```diff
@@ -376,17 +391,24 @@ func (we *watcherEvents) checkStateChanges(oldState, newState *types.TipSet) map
 	defer we.lk.RUnlock()
 
 	res := make(map[triggerID]eventData)
+	wg := sync.WaitGroup{}
 	for tid, matchFn := range we.matchers {
-		ok, data, err := matchFn(oldState, newState)
-		if err != nil {
-			log.Errorf("event diff fn failed: %s", err)
-			continue
-		}
+		wg.Add(1)
+		go func(tid uint64, matchFn StateMatchFunc) {
+			ok, data, err := matchFn(oldState, newState)
+			if err != nil {
+				log.Errorf("event diff fn failed: %s", err)
+				wg.Done()
+				return
+			}
 
-		if ok {
-			res[tid] = data
-		}
+			if ok {
+				res[tid] = data
```

Comment:
This map access needs a lock

Reply:
could use an array instead of a map and lock